### DISCUSSION TITLE
How to magit-commit-create without diff poping up on other window.

### DISCUSSION THREAD

**Discussion by paxsali** [magit/magit#4336] [Support] [answered]
Hi all,

Before I commit, I always check the diffs ahead of time. In other words, I never commit without knowing what I am doing, therefore I don't want to see no diff **while** I'm writing the commit message.

When I stage files and then attempt to commit them, magit opens a diff buffer in "other window".

I don't want that. I prefer to simply enter the commit message in the current window (that's how I setup my magit) and that's it.

The option `-v Show diff of changes to be committed (--verbose)` is OFF all the time, btw.

My magit config section is this here (only relevant parts are posted):

```
;;;; * Magit
  (with-eval-after-load 'magit
;;;;; * Magit functions
    ;; Source:
    ;; https://idiomdrottning.org/magit-transients
    ;; How to put an end to Magit and its wonderful popups
    (defun magit-display-buffer-same-window (buffer)
      "Display BUFFER in the selected window like God intended."
      (display-buffer buffer '(display-buffer-same-window)))
;;;;; * Magit settings
    (setq magit-display-buffer-function 'magit-display-buffer-same-window)
    (setq magit-popup-display-buffer-action '((display-buffer-same-window)))
;;;;; * Magit keybindings
    (define-key magit-mode-map (kbd "R") 'magit-file-rename)
    )
```

How can I achieve this?

Thanks for your suggestion in advance.

**Comment by tarsius** [comment]
You can disable that using

```elisp
(with-eval-after-load 'magit-commit
  (remove-hook 'server-switch-hook 'magit-commit-diff))
```

**Comment by kyleam** [comment, marked as answer]
There's also the `magit-commit-show-diff` option.  When set to `nil`, it makes `magit-commit-diff` exit right away, achieving the same effect as removing `magit-commit-diff` from `server-switch-hook` at the price of a pointless function call.

If `magit-commit-diff` is dropped from `server-switch-hook` while keeping `magit-commit-show-diff` at its default value of `t`, commands that use `magit-commit-squash-internal` (`magit-commit-squash`, `magit-commit-fixup`, ...) still show a diff.

**Reply by tarsius** [reply]
We should mention the option in the manual. I've added an entry to my private todo list.

**Reply by tarsius** [reply]
Done. :)

**Reply by paxsali** [reply]
Thank you, guys. `magit-commit-show-diff` did the trick for me.
